openproblems/tasks/_batch_integration/batch_integration_embed/methods/harmony.py

```diff
@@ -0,0 +1,96 @@
+# from ....tools.normalize import log_cpm
+from .....tools.decorators import method
+from .....tools.utils import check_version
+
+import scprep
+
+# from scIB.integration import _harmony
+
+
+_harmony = scprep.run.RFunction(
+    setup="""
+            library(SingleCellExperiment)
+            library(harmony)
+            library(Seurat)
+        """,
+    args="sobj, batch",
+    body="""
+            seu <- as.Seurat(sobj)
+            seu <- ScaleData(seu)
+            seu <- RunPCA(seu, features=rownames(seu@assays$RNA))
+            seu <- RunHarmony(seu, batch)
+            seu <- as.SingleCellExperiment(seu)
+            seu
+        """,
+)
+
+
+@method(
+    method_name="Harmony",
+    paper_name="Sc",
+    paper_url="temp",
+    paper_year=2020,
+    code_url="",
+    code_version=check_version("scprep"),
+    image="openproblems-r-extras",  # only if required
+)
+def harmony_full_unscaled(adata):
+    corrected = _harmony(adata, "batch")
+    adata.obsm["X_emb"] = corrected.obsm["HARMONY"]
+    return adata
```

Comment:
Why not import from the `_graph` function? You can add the `reduce_data` call...